### PR TITLE
feat: Educator Page

### PR DESCRIPTION
**Pull request by MAbdurrehman12**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
https://github.com/mitodl/ocw-hugo-themes/issues/307

#### What's this PR do?
Educator's page created which can be accessed from homepage, by clicking "Educators start here" 

#### How should this be manually tested?

- This PR is Deployed [here](https://ocw-hugo-themes-pr-309--ocw-next.netlify.app/educator/)
- Clicking "Educators Start Here" in homepage should redirect to this educator page.
- One long, scrolling page
- on-page nav should stay sticky, at top of viewport when user scrolls down 
- Clicking on item in on-page nav should smooth-scroll down to the appropriate section, not jump.
- Verify that all links work fine
- Should be a part of ocw-www theme
- Compare with design: https://invis.io/7Y12598HHF6E#/462148864_Educators_Start_Here
- Proofread content: https://docs.google.com/document/d/1DTD0cHXB_n3UH1HZwyD8adFOfrgxfY_zSzY5pTTyKyE
- As much as possible, styles should follow existing styles (e.g. fonts & colors), especially those added in About Us Page #303
- Test the responsiveness of the page for various screen sizes, including:
    - 1920×1080
    - 1366×768 
    - 768×1024 
    - 414×816 
    - 375×812
- Verify that the embedded video play
- Test the page on multiple browsers"
- Any other instructions, mentioned in relevant ticket

#### Screenshots (if appropriate)
Laptop screen: 

educator page SS

Mobile screen:

mobile SS

### CODE REVIEW

**Comment by MAbdurrehman12** [comment]
Currently, I have redirected all the **non ocw.mit.edu links** in new tab and all the **ocw.mit.edu links** in the same tab.
Is there any convention to follow or it is fine this way? 

Should we redirect this to Chalk Radio Podcast youtube or there's any other link?
image

**Comment by pdpinch** [comment]
I'm concerned that the CSS is getting too complicated. Can we use the same fonts that @gumaerc used in #303 ?

**Comment by MAbdurrehman12** [comment]
> I'm concerned that the CSS is getting too complicated. Can we use the same fonts that @gumaerc used in #303 ?

Are you referring to font family, color, or size?
I have used the same font family and color that Carey used, for font size, I have taken that from the PSD.
Do you also want me to use the same font size which Carey used or there's anything else that I'm missing?

ps: I have also attached a screenshot above

**Comment by pdpinch** [comment]
I was mostly thinking of font family. Carey's PR hasn't merged yet, so the CSS diff looks bigger than it should. 

I will take a closer look at the font sizes, but please look for any opportunities to consolidate and simplify the CSS.

**Comment by MAbdurrehman12** [comment]
> I was mostly thinking of font family. Carey's PR hasn't merged yet, so the CSS diff looks bigger than it should. I will take a closer look at the font sizes, but please look for any opportunities to consolidate and simplify the CSS.

So before starting work, I checked out a new branch from Carey's branch, hence I got his fonts file (base-theme/assets/css/font.scss) and then rebased my branch with main. If there's any changes then we can handle them before merging.

Sure, I'll simplify the CSS. 
The css which I have written for educator's page is here: [www/assets/css/educator.scss](https://github.com/mitodl/ocw-hugo-themes/pull/309/files#diff-2fedd92c1d22a4427396f74910d4b6cb3f9033b9879ac56f0fef0e039aab9f15).
Are you referring to the same css?

**Comment by Wassaf-Shahzad** [comment]
There are some minor changes
- Line height for the first block of text under `Welcome Educators` should be 35px
- Colour for links should be #126F9A 
- Links displayed in lists are using bold version Helvetica, you could probably increase the font weight for it
- Line height of the block of text under `Explore Open Reflective Practice at MIT` should be 22px
- The text below `Get motivated ` heading should not be bold and the spacing between the heading and body of text should be around 13px

**Review by Wassaf-Shahzad — CHANGES_REQUESTED** [review]
I added a note for the css changes

**Comment by MAbdurrehman12** [comment]
> I added a note for the css changes

Thanks, I'll look into them

**Comment by pdpinch** [comment]
The screenshots look OK to me. We'll probably get (at least) another round of feedback after this is deployed, but I'd like to get it out the door.

File: www/assets/js/subnav.js [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
+  $(window).on("scroll", function() {
+    const position = window.pageYOffset
+    $(".section").each(function() {
+      const target = $(this).offset().top
+      const navHeight = $(".on-page-sub-nav").height()
+      if (position >= target - navHeight - 50) {
+        const navLinks = $(".on-page-sub-nav  li a")
+        const id = $(this).attr("id")
+        navLinks.removeClass("active")
+        $(`.on-page-sub-nav  li a[href="#${id}"]`).addClass("active")
```
Comment (Wassaf-Shahzad) [inline www/assets/js/subnav.js:29]: This code currently adds active class to the suitable nav-item on scroll. However if you view this in mobile view the nav-brand does not change to the suitable  nav-item text on scroll, it remains the default one or the first one.However on opening the nav menu and clicking nav-item it doesn't change the nav-brand

```
const activeElement = $(`.on-page-sub-nav  li a[href="#${id}"]`)
activeElement.addClass("active")
const brandText = activeElement.text()
$(".navbar-brand > span").text(brandText)
```

The above code can resolve this issue after that feel free to merge

**Review by Wassaf-Shahzad — CHANGES_REQUESTED** [review]

File: www/assets/js/subnav.js (hunk 1 above)
Comment (MAbdurrehman12) [reply www/assets/js/subnav.js]: Thanks for pointing it out Wassaf!
I've changed it, please have a look.

**Comment by MAbdurrehman12** [comment]
Note: Currently we are using window.scrollTo when user clicks on a nav item and we are also changing nav brand when user scrolls so there is a little glitch when we user clicks on a nav item as window.scrollTo function is triggered which also changes the nav brand quite rapidly, turning out to be a glitch.

This issue can be observed when deployed and rectified with other changes accordingly.

**Review by Wassaf-Shahzad — APPROVED** [review]
LGTM

**Pull request merged**